src/components/decimal/decimal.test.tsx

```diff
@@ -0,0 +1,1406 @@
+import React, { useState } from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import Decimal from "./decimal.component";
+import type { DecimalProps, CustomEvent } from "./decimal.component";
+import { testStyledSystemMargin } from "../../__spec_helper__/__internal__/test-utils";
+import FormFieldStyle from "../../__internal__/form-field/form-field.style";
+import I18nProvider from "../i18n-provider";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
+
+const ControlledDecimal = ({
+  onChange,
+  startingValue,
+  ...rest
+}: Partial<DecimalProps> & { startingValue?: string }) => {
+  const [value, setValue] = useState(startingValue);
+  const handleChange = (e: CustomEvent) => {
+    setValue(e.target.value.rawValue);
+    onChange?.(e);
+  };
+
+  return <Decimal value={value} onChange={handleChange} {...rest} />;
+};
+
+testStyledSystemMargin(
+  (props) => <Decimal {...props} />,
+  undefined,
+  (component) => component.find(FormFieldStyle),
+  { modifier: "&&&" }
+);
+
+describe("when the component is uncontrolled", () => {
+  it("displays a deprecation warning", () => {
+    const loggerSpy = jest.spyOn(Logger, "deprecate");
+    render(<Decimal defaultValue="0.01" />);
+
+    expect(loggerSpy).toHaveBeenCalledWith(
+      "Uncontrolled behaviour in `Decimal` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+    );
+    expect(loggerSpy).toHaveBeenCalledTimes(1);
+
+    loggerSpy.mockRestore();
+  });
+
+  it("has a default value of 0.00 when no defaultValue prop is provided", () => {
+    render(<Decimal />);
+    expect(screen.getByRole("textbox")).toHaveValue("0.00");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("0.00");
```

Comment:
```suggestion
    expect(screen.getByRole("textbox", { hidden: true })).toHaveValue("0.00");
```

Reply:
as discussed privately, this change doesn't work - it still picks up the visible input not the hidden one (the `{hidden: true}` just modifies the query so it can also pick up any text inputs that are hidden from the accessibility tree through eg `aria-hidden` - but an `<input type="hidden" />` doesn't even have the textbox role so the query doesn'g find it). As a result, although the change works on this particular test it only does so because both inputs have the same value (and it's just repeating the assertion on the line above), it fails on many other tests.